Update NotEnoughReputation component

ArmandoGraterol opened the pull request:
Updated NotEnoughtReputation component copies and styles

**New stuff** ✨

* Added an appearance prop to NotEnoughReputation in order to control when to apply the negative margin top

**Changes** 🏗

* Updated copies of NotEnoughReputation

![FireShot Capture 354 - Colony - localhost](https://user-images.githubusercontent.com/18473896/122264827-63dd2200-ceae-11eb-8b13-9931f25e73e7.png)

Resolves DEV-410

alicjakujawa reviewed (changes requested):
Firstly, are we sure that this is correct design?
<img width="498" alt="Screenshot 2021-06-17 at 10 03 26" src="https://user-images.githubusercontent.com/13069555/122356680-4f069a00-cf53-11eb-9e00-c07457b6f43b.png">

Secondly, is this component use somewhere where it should have this margin? 
And lastly, shouldn't we have this component in "base" state as component with margin, and adding margin (not negative margin) by appearance object?  Unless this component is displayed everywhere same, then we dont need to appearance object at all

chinins commented on src/modules/dashboard/components/NotEnoughReputation/NotEnoughReputation.tsx:
```diff
@@ -1,37 +1,54 @@
-      'There is not enough reputation in colony/domain to create this motion.',
+    defaultMessage: `There is no reputation in this {onRootDomain, select,
+      true {colony}
+      false {domain}
```
Should be `team` here as we don't use domains in the copies.

rdig replied:
Good catch! :1st_place_medal:

chinins reviewed (approved):
@alicjakujawa, the designs are what Karol provided at the time (pink title being a different color than the permissions warning). I've just had a call with him on a different matter and he confirmed that we can change the color to the red that is used for warnings.

As for the margin in the `NotEnoughReputation`, the margin comes from the dialog section. Depending on the dialog, we don't need the negative margin, so it is sometimes needed and sometimes not. So it's either we conditionally add/remove margin on the dialog section of the above part and also add/remove the margin-top of that above part as well, or we do a negative margin prop on the component. Another option would be to wrap it in a div & apply the negative margin on that div. Not sure which one is the best. Maybe the prop as it doesn't add another div level.

So I'm pre-approving, just please change the color of the red title & correct `domain` to `team` in the copy.

ArmandoGraterol commented:
@alicjakujawa 

1. From what I know, I think so. A screenshot of the design was added on the notion issue and Karol posted a link to the design.
2. If you're talking about the negative margin, yes. The negative margin is used on 4 instances where the component is used.
3. The margin that you're probably thinking about comes from the DialogSections that are above this component because apart of the negative margin I just added, the NotEnoughReputation doesn't any.

Sometimes the DialogSections above the NotEnoughReputation component have a lot of margin/padding on the bottom (It depends on the Dialog), the NotEnoughReputation component is used in 8 dialogs, in which 4 of them a negative margin is needed in order to compensate for the larger margin/padding of the sections.

I didn't want to add another div to the component, so that's why I changed the position of the previous container and added an appearance object for the cases where the negative margin is needed.

Let me know what you think, and if that can be done in a better way
